From a40064979706dd4f184a07d9dfb6fefffeaf090b Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Tue, 11 Mar 2014 11:17:53 -0400
Subject: [PATCH 177/225] Ticket 47722 - rsearch filter error on any search
 filter

Bug Description:  A NULL gets incorrectly added to the search filter, which
                  creates an invalid filter.

Fix Description:  Do not add a NULL character to the search filter.

Original fix submitted by nkinder(Thanks!)

https://fedorahosted.org/389/ticket/47722

Reviewed by: nhoosi(Thanks!)
(cherry picked from commit 8afcf90b0d9d6599144bb1666ca79cef9e91b3ef)
(cherry picked from commit 0aa38a22b02b3e6ad486711424dc74f2d09662c3)
---
 ldap/servers/slapd/tools/rsearch/searchthread.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ldap/servers/slapd/tools/rsearch/searchthread.c b/ldap/servers/slapd/tools/rsearch/searchthread.c
index 8a74d58..e92209f 100644
--- a/ldap/servers/slapd/tools/rsearch/searchthread.c
+++ b/ldap/servers/slapd/tools/rsearch/searchthread.c
@@ -387,7 +387,7 @@ static int st_search(SearchThread *st)
             sprintf(num, "%d", get_large_random_number() % numeric);
             s = num;
         }
-        sprintf(filterBuffer, "%s%s", filter, s);
+        sprintf(filterBuffer, "%s%s", filter, s ? s : "");
         pFilter = filterBuffer;
     } else {
         pFilter = filter;
-- 
1.8.1.4

